lib/mix/test/mix/tasks/xref_test.exs

```diff
@@ -253,59 +261,96 @@ defmodule Mix.Tasks.XrefTest do
 
     """
 
-    assert_warnings(code, warning)
+    assert_warnings(files, warning)
   end
 
   test "warnings: doesn't report missing funcs at compile time" do
-    assert_no_warnings("""
-      Enum.map([], fn _ -> BadReferencer.no_func4() end)
+    files = %{
+      "lib/a.ex" => """
+        Enum.map([], fn _ -> BadReferencer.no_func4() end)
+
+        if function_exported?(List, :flatten, 1) do
+          List.flatten([1, 2, 3])
+        else
+          List.old_flatten([1, 2, 3])
+        end
+      """
+    }
+    assert_no_warnings(files)
+  end
 
-      if function_exported?(List, :flatten, 1) do
-        List.flatten([1, 2, 3])
-      else
-        List.old_flatten([1, 2, 3])
+  test "warnings: reports deprecated calls" do
+    files = %{
+      "lib/a.ex" => """
+      defmodule A do
+        @deprecated "Use A.c/0 instead"
+        def a, do: c()
+
+        def b, do: a()
```

Comment:
There is a problem here, that I can't warn for local files since they aren't in the manifest. Should I transverse the definitions or add a warning on `elixir_locals`?

Reply:
Let's document we won't warn for local calls... only the qualified ones.